ompi/mca/coll/basic/coll_basic_alltoall.c

```diff
@@ -102,6 +99,7 @@ mca_coll_basic_alltoall_inter(const void *sbuf, int scount,
                                  MCA_COLL_BASE_TAG_ALLTOALL,
                                  MCA_PML_BASE_SEND_STANDARD, comm, sreq));
         if (OMPI_SUCCESS != err) {
+            ompi_coll_base_free_reqs(req, nreqs);
```

Comment:
@bosilca   Replace
    ompi_coll_base_free_reqs(req, nreqs);
by
    ompi_coll_base_free_reqs(req, size+i+1);
No need to free nreqs requests since only size+i+1 of them have been filled